core-plugins/src/test/java/co/cask/hydrator/plugin/batch/aggregator/RowDenormalizerAggregatorTest.java

```diff
@@ -0,0 +1,509 @@
+
+
+    ETLStage sinkStage = new ETLStage(
+      "sink", new ETLPlugin("TPFSAvro", BatchSink.PLUGIN_TYPE,
+                            ImmutableMap.of(Properties.TimePartitionedFileSetDataset.SCHEMA, outputSchema.toString(),
+                                            Properties.TimePartitionedFileSetDataset.TPFS_NAME, outputDatasetName),
+                            null));
+
+    ETLBatchConfig config = ETLBatchConfig.builder("* * * * *")
+      .addStage(sourceStage)
+      .addStage(aggregateStage)
+      .addStage(sinkStage)
+      .addConnection(sourceStage.getName(), aggregateStage.getName())
+      .addConnection(aggregateStage.getName(), sinkStage.getName())
+      .build();
+    AppRequest<ETLBatchConfig> request = new AppRequest<>(DATAPIPELINE_ARTIFACT, config);
+    Id.Application appId = Id.Application.from(Id.Namespace.DEFAULT, "denormalize-test");
+    ApplicationManager appManager = deployApplication(appId, request);
+
+    // write input data
+    DataSetManager<Table> inputManager = getDataset(inputDatasetName);
+    Table inputTable = inputManager.get();
+    Put put = new Put(Bytes.toBytes(1));
+    put.add("KeyField", "A");
+    put.add("FieldName", "Firstname");
+    put.add("FieldValue", "ABC");
+    inputTable.put(put);
+    put = new Put(Bytes.toBytes(2));
+    put.add("KeyField", "A");
+    put.add("FieldName", "Lastname");
+    put.add("FieldValue", "XYZ");
+    inputTable.put(put);
+    put = new Put(Bytes.toBytes(3));
+    put.add("KeyField", "A");
+    put.add("FieldName", "Address");
+    put.add("FieldValue", "PQR place near XYZ");
+    inputTable.put(put);
+    inputManager.flush();
+
+    // run the pipeline
+    WorkflowManager workflowManager = appManager.getWorkflowManager(SmartWorkflow.NAME);
+    workflowManager.start();
+    workflowManager.waitForFinish(5, TimeUnit.MINUTES);
+
+    DataSetManager<TimePartitionedFileSet> outputManager = getDataset(outputDatasetName);
+    TimePartitionedFileSet fileSet = outputManager.get();
+    List<GenericRecord> outputRecords = readOutput(fileSet, outputSchema);
+
+    Set<String> key = new HashSet<>();
+    Set<String> firstName = new HashSet<>();
+    Set<String> lastName = new HashSet<>();
+    Set<String> address = new HashSet<>();
+
+    for (GenericRecord record : outputRecords) {
+      key.add(record.get("KeyField").toString());
+      firstName.add(record.get("Firstname").toString());
+      lastName.add(record.get("Lastname").toString());
+      address.add(record.get("Address").toString());
+    }
+
+    Assert.assertEquals(1, outputRecords.size());
+    Assert.assertEquals(ImmutableSet.of("A"), key);
+    Assert.assertEquals(ImmutableSet.of("ABC"), firstName);
+    Assert.assertEquals(ImmutableSet.of("XYZ"), lastName);
+    Assert.assertEquals(ImmutableSet.of("PQR place near XYZ"), address);
+  }
+
+  @Test
+  public void testDenormalizerWithSelectedAlias() throws Exception {
```

Comment:
this test seems to be a superset of testDenormalizerWithAllFields. If so, there is no point in having the other test.